src/molecules/Carousel/index.js

```diff
@@ -0,0 +1,253 @@
+					a.addEventListener('click', (e) => {
+						// eslint-disable-next-line react/destructuring-assignment
+						if (this.state.isSwiping) {
+							e.preventDefault();
+						}
+					});
+				});
+				this.setCarouselLength(carouselContentEl);
+
+				// recalculating dimensions onresize
+				window.addEventListener('resize', () => this.setCarouselLength(carouselContentEl));
+			}
+		);
+	}
+
+	setCarouselLength(carouselContentDom) {
+		const dimensions = carouselContentDom.getBoundingClientRect();
+
+		const { vertical } = this.props;
+
+		this.setState({
+			carouselLength: vertical ? carouselContentDom.scrollHeight : carouselContentDom.scrollWidth,
+			pageLength: vertical ? dimensions.height : dimensions.width,
+		});
+	}
+
+	get enoughItemsForScroll() {
+		const { carouselLength, pageLength } = this.state;
+
+		return pageLength < carouselLength;
+	}
+
+	move(delta) {
+		const { vertical } = this.props;
+		const { carouselContentEl } = this.state;
+
+		if (vertical) {
+			carouselContentEl.scrollTop += delta;
+		} else {
+			carouselContentEl.scrollLeft += delta;
+		}
+	}
+
+	doSliding(direction) {
+		this.setState(
+			{
+				pageIsTurning: true,
+			},
+			() => {
+				this.swipePage(direction);
+
+				window.setTimeout(() => {
+					this.setState({
+						pageIsTurning: false,
+					});
+				}, 400);
+			}
+		);
+	}
+
+	swipePage(direction) {
+		const { pageLength } = this.state;
+
+		const delta = direction === 'next' ? pageLength : -pageLength;
+		this.move(delta);
+	}
+
+	dragFinished() {
+		const { withMouseEvents } = this.props;
+
+		if (!withMouseEvents) {
+			return;
+		}
+
+		// disable link 'click' after swipe in Chrome
+		window.setTimeout(() => {
+			this.setState({
+				isSwiping: false,
+			});
+		});
+	}
+
+	dragStarted(e) {
+		const { withMouseEvents } = this.props;
+
+		if (!withMouseEvents) {
+			return;
+		}
+
+		this.setState({
+			mouseX: e.clientX,
+			mouseY: e.clientY,
+		});
+	}
+
+	drag(e) {
+		const { withMouseEvents, vertical } = this.props;
+
+		if (!withMouseEvents) {
+			return;
+		}
+
+		const { mouseX, mouseY } = this.state;
+		const { clientX, clientY, buttons } = e;
+
+		// checking if drag is performed with pressed mouse button
+		if (!buttons) {
+			return;
+		}
+
+		const deltaX = mouseX - clientX;
+		const deltaY = mouseY - clientY;
+
+		const delta = vertical ? deltaY : deltaX;
+
+		this.move(delta);
+
+		this.setState({
+			mouseX: clientX,
+			mouseY: clientY,
+			isSwiping: true,
+		});
+	}
+
+	render() {
+		const {
+			label, children, vertical,
+		} = this.props;
+		const {
+			direction, pageIsTurning, prevArrowVisible, nextArrowVisible,
+		} = this.state;
+		return (
+			<Fragment>
+				{label}
+				<Wrapper {...this.props}>
+					<CarouselContainer
+						direction={direction}
+						pageIsTurning={pageIsTurning}
+						onScroll={this.onScroll}
+						{...this.props}
+					>
+						{/* div instead of styled-component because we're passing ref */}
+						<div
```

Comment:
Would it be possible to combine this div with the surrounding `CarouselContainer`?